go/plugins/dotprompt/genkit.go

```diff
@@ -216,10 +236,106 @@ func (p *Prompt) Generate(ctx context.Context, pr *PromptRequest, cb func(contex
 		}
 	}
 
-	resp, err := model.Generate(ctx, genReq, cb)
+	resp, err := model.Generate(ctx, mr, pr.Stream)
+	if err != nil {
+		return nil, err
+	}
+
+	return resp, nil
+}
+
+// GenerateText runs generate request for this prompt. Returns generated text only.
+func (p *Prompt) GenerateText(ctx context.Context, opts ...GenerateOption) (string, error) {
+	res, err := p.Generate(ctx, opts...)
+	if err != nil {
+		return "", err
+	}
+
+	return res.Text(), nil
+}
+
+// GenerateData runs generate request for this prompt. Returns ModelResponse struct.
+// TODO: Stream GenerateData with partial JSON
+func (p *Prompt) GenerateData(ctx context.Context, value any, opts ...GenerateOption) (*ai.ModelResponse, error) {
+	with := WithOutputType(value)
```

Comment:
Nit: Can just append `WithOutputType` to opts and let it be handled later so this isn't done in multiple places? Would be nice to throw a custom error if doing `GenerateData` + manually passing `WithOutputType` (user might be confused because they didn't set output type twice) but I don't think there's a succinct way to do that...